pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -220,50 +235,97 @@ func (w *watcher) updateValues(ctx context.Context, repo git.Repo, repoID string
 	}
 	defer tmpRepo.Clean()
 
-	newEvents := make([]model.Event, 0, len(eventCfgs))
+	milestone, ok := w.milestoneMap.Load(repoID)
+	if !ok {
+		milestone = 0
+	}
+	var (
+		firstRead      = !ok
+		handledEvents  = make([]*pipedservice.ReportEventStatusesRequest_Event, 0, len(eventCfgs))
+		outDatedEvents = make([]*pipedservice.ReportEventStatusesRequest_Event, 0)
+		maxTimestamp   int64
+	)
 	for _, e := range eventCfgs {
-		latestEvent, ok := w.eventStore.GetLatest(ctx, e.Name, e.Labels)
-		if !ok {
+		notHandledEvents := w.eventLister.ListNotHandled(e.Name, e.Labels, milestone.(int64)+1, numToMakeOutdated)
+		if len(notHandledEvents) == 0 {
 			continue
 		}
-		if latestEvent.IsHandled() {
-			continue
+		if len(notHandledEvents) > 1 {
+			// Events other than the latest will be OUTDATED.
+			for i := 1; i < len(notHandledEvents[1:]); i++ {
```

Comment:
```suggestion
			for i := 0; i < len(notHandledEvents[1:]); i++ {
```

Reply:
Basically, the first element is the latest event, so it shouldn't be OUTDATE. But on the first read, it can happen. So let me update `outDatedEvents` within the `if firstRead` block.

Reply:
>  the first element is the latest event, so it shouldn't be OUTDATE

Yes, I see, but the code is skipping 2 elements: the first one and the last one.

Reply:
For sure! So let me apply your:
```go
for _, e := range notHandledEvents[1:]
```